bot/bot.py

```diff
@@ -1,62 +1,38 @@
-import asyncio
-import socket
-from typing import Optional
-
-import aiohttp
 import discord
-from async_rediscache import RedisSession
+from botcore import BotBase
 from botcore.site_api import APIClient
-from botcore.utils.extensions import walk_extensions
 from botcore.utils.logging import get_logger
 from botcore.utils.scheduling import create_task
-from discord.ext import commands
 
 from bot import constants, exts
 from bot.exts.code_jams._views import JamTeamInfoView
 
 log = get_logger(__name__)
 
 
-class SirRobin(commands.Bot):
+class SirRobin(BotBase):
     """Sir-Robin core."""
 
-    def __init__(self, redis_session: RedisSession, **kwargs):
-        super().__init__(**kwargs)
-
-        # This session may want to be recreated on login/disconnect.
-        # Additionally, on the bot repo we use a different connector that is more "stable".
-        self.code_jam_mgmt_api: Optional[APIClient] = None
-        self.http_session: Optional[aiohttp.ClientSession] = None
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
 
-        self._guild_available: Optional[asyncio.Event] = None
-        self.redis_session = redis_session
+        self.code_jam_mgmt_api: APIClient = None
```

Comment:
```suggestion
        self.code_jam_mgmt_api: APIClient | None = None
```

The change from `Optional[APIClient]` to `APIClient` would appear to make this type hint incorrect, as it's being initialized with`None`